CMakeLists.txt

```diff
@@ -147,7 +147,11 @@ set(DCA_LIBS
   cuda_utils
 )
 
+set(SYSTEM_GPU_COUNT 0)
+
 if (DCA_HAVE_CUDA)
+  EXECUTE_PROCESS(COMMAND bash -c "nvidia-smi -L | awk 'BEGIN { num_gpu=0;} /GPU/ { num_gpu++;} END { printf(\"%d\", num_gpu) }'"
```

Comment:
The ring algorithm should be independendent from the numbner of GPUs on the node. Rather the technical issue is that we rely on cuda aware MPI. Can we have a check similar to hostname==summit for the moment? Assuming automatically detecting if MPI is cuda aware is more complicated

Reply:
The cvd script will definitely have undefined behavior if there is not 1 GPU per rank.

Reply:
In FAQ for Open-MPI, there is mention of "Can I tell at compile time or runtime whether I have CUDA-aware support?" . It seems to test for MPIX_CUDA_AWARE_SUPPORT
https://www.open-mpi.org/faq/?category=runcuda#mpi-cuda-dev

Just FYI

Reply:
Added #210 and with that I think this should go in so #208 and #206 can pass CI and go in.